micro-rdk/src/common/robot.rs

```diff
@@ -87,12 +101,22 @@ impl ResourceType {
     }
 }
 
+#[cfg(feature = "native")]
```

Comment:
Could we combine these with L.11 a la `type Executor = crate::<esp32/native>::exec::<executor_name>`?

Reply:
https://viam.atlassian.net/browse/RSDK-8123